app/models/user.rb

```diff
@@ -1,6 +1,64 @@
 class User < ApplicationRecord
   # Include default devise modules. Others available are:
-  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
-  devise :database_authenticatable, :registerable,
-         :recoverable, :rememberable, :validatable
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable, :registerable,
+  devise :invitable, :database_authenticatable,
+         :recoverable, :rememberable, :validatable, invite_for: 2.weeks
+
+  has_many :taught_classes, dependent: :destroy
+  has_many :assignments, through: :taught_classes
+  has_many :class_memberships, dependent: :destroy
+  has_many :student_answers, through: :completed_assignments
+  has_many :completed_assignments
+
+  belongs_to :school
+
+
+  enum role: [:teacher, :student, :admin]
```

Comment:
I would prefer an enum hash instead of an array. (https://naturaily.com/blog/ruby-on-rails-enum,  only 1.Declare enum as a Hash not Array)